File: 02_Two_Pointers/01_283. Move Zeroes.cpp
// Problem Link -> https://leetcode.com/problems/move-zeroes/description/

// ------------------------------- Approach - 1 ---------------------------

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        vector<int> tempArr;
        for(int i = 0; i < nums.size(); i++){
            if(nums[i] != 0){
                tempArr.push_back(nums[i]);
            }
        }

        for(int i = 0; i < tempArr.size(); i++){
            nums[i] = tempArr[i];
        }

        for(int i = tempArr.size(); i < nums.size(); i++){
            nums[i] = 0;
        }
    }
};



// Time Complexity -> O(n)
// Space Complexity -> O(n)

File: 02_Two_Pointers/01_283. Move Zeroes.py
# Problem Link -> https://leetcode.com/problems/move-zeroes/description/

# ------------------------------- Approach - 1 ---------------------------

class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        temp_list = []
        for num in nums:
            if num != 0:
                temp_list.append(num)
        
        for i in range(len(temp_list)):
            nums[i] = temp_list[i]
        
        for i in range(len(temp_list), len(nums)):
            nums[i] = 0



# Time Complexity -> O(n)
# Space Complexity -> O(n)

# ------------------------------------------------------------------------
